### PR TITLE
Changed Cores to use a Double value instead of a Float for convenience.

### PR DESCRIPTION
**Pull request by tfenne**
This one's for you @nh13

### CODE REVIEW

**Comment by tfenne** [comment]
Added a second commit with some scala magic (and nasty recursive generics) that allow us to define mathematical operations on Resource that work for all subclasses.

File: core/src/main/scala/dagr/core/execsystem/ResourceSet.scala [hunk 1]
```diff
@@ -25,7 +25,7 @@ package dagr.core.execsystem
 
 object ResourceSet {
   def apply(that: ResourceSet): ResourceSet = new ResourceSet(cores = that.cores, memory = that.memory)
-  def apply(cores: Float, memory: Long) = new ResourceSet(Cores(cores), Memory(memory))
+  def apply(cores: Double, memory: Long) = new ResourceSet(Cores(cores), Memory(memory))
```
Comment (nh13) [inline core/src/main/scala/dagr/core/execsystem/ResourceSet.scala:28]: Line 31: `Float.MaxValue` -> `Double.MaxValue`

**Comment by nh13** [comment]
A few more places: 
- https://github.com/fulcrumgenomics/dagr/blob/4cadb0d5e75e64c75ff0db897c65f28d346e89d6/core/src/main/scala/dagr/core/cmdline/DagrCoreMain.scala#L95
- https://github.com/fulcrumgenomics/dagr/blob/4cadb0d5e75e64c75ff0db897c65f28d346e89d6/core/src/main/scala/dagr/core/cmdline/DagrCoreMain.scala#L126
- https://github.com/fulcrumgenomics/dagr/blob/4cadb0d5e75e64c75ff0db897c65f28d346e89d6/core/src/main/scala/dagr/core/execsystem/TaskManager.scala#L43

**Comment by nh13** [comment]
@tfenne back to you

**Comment by tfenne** [comment]
Fixed all remaining Floats related to Cores, and squashed the commits.  @nh13 want to re-review or merge?

**Pull request merged**